### DISCUSSION TITLE
The new version of ORM prompts an issue in pycharm

### DISCUSSION THREAD

**Discussion by breezechime** [sqlalchemy/sqlalchemy#11265] [Usage Questions]
### Describe the bug

In versions less than 2.0, the __ init__ method of the ORM model is normally prompted by pychar, and the corresponding parameters of the ORM model can be filled in. Even if the __ init__ method is overwritten, there will be a prompt.
But in versions higher than 2.0, the _ init__ method of the ORM model cannot be prompted by pychar and vs code to indicate the parameters that the ORM model can fill in.

ZH：在小于2.0的版本中ORM模型的__init__方法正常被pycharm提示，可填写ORM模型对应的参数，即使__init__方法被覆盖也会有提示。
但在高于2.0的版本中ORM模型的__init__方法无法被pycharm和vs code提示ORM模型可填的参数。

this is old version normal tips
![a806c6fe5ddf4da8c853943f68ad45f](https://github.com/sqlalchemy/sqlalchemy/assets/89468384/930082f8-c152-4377-8166-84707fd03cdb)

higher than 2.0 version question, tips error
![cf664d87c0a6493c68c133d187063d6](https://github.com/sqlalchemy/sqlalchemy/assets/89468384/8b324afa-4b51-46f7-8a3d-1dd0c51356da)

### SQLAlchemy Version in Use

2.0.29

### DBAPI (i.e. the database driver)

pymysql

### Database Vendor and Major Version

mysql8

### Python Version

python3.11、python3.9

### Operating system

Windows

### To Reproduce

```python
from sqlalchemy.orm import declarative_base

Base = declarative_base()


class Test(Base):
    __tablename__ = 'test'

    id = Column(BIGINT, primary_key=True)
    name = Column(String(255))

    def __init__(self):
        pass


Test()
```

### Error

```
# Copy the complete stack trace and error message here, including SQL log output if applicable.
```

**Comment by CaselIT** [comment]
Hi,

From sqlalchemy point of view the default init has always been typed as `**kwargs`.

This seems something that's related to pycharm, and you should probably seek guidance on their support forum.

In version 2.0 yoi can use dataclasses to generate an init that woll be typed

**Reply by breezechime** [reply]
Does your ORM model have a __ init__ prompt?

**Reply by CaselIT** [reply]
sorry I think I miss understood your question.

That behavior seems to be a bug in pycharm. It's type checker does not appear to be very good, since there are also some other wired things that it does.

In any case it seems like the issue was on 1.4 not on 2.0 so not sure why this was opened as an issue. I don't see something actionable on sqlalchemy side
